### PR TITLE
[TypeChecker] Enable result builder AST transform by default

### PR DESCRIPTION
**Pull request by xedin**
This PR:

- Drops experimental feature flag (`ResultBuilderASTTransform`)
- Adjusts code completion (changes by @ahoppen)
- Adjusts all the improved test-cases
- Removes original result builder implementation

Resolves: rdar://89880662
Resolves: rdar://97388257
Resolves: rdar://95235303
Resolves: rdar://96343005
Resolves: rdar://91731420
Resolves: rdar://95977761
Resolves: rdar://80680636

### CODE REVIEW

File: include/swift/Sema/ConstraintSystem.h [hunk 1]
```diff
@@ -6245,6 +6245,10 @@ class ConjunctionElement {
     Element->print(Out, SM, indent);
   }
 
+  /// Returns \c false if this conjunction element is known not to contain the
+  /// code compleiton token.
```
Comment (LucianoPAlmeida) [inline include/swift/Sema/ConstraintSystem.h:6249]: Typo?
```suggestion
  /// code completion token.
```

Comment (xedin) [reply include/swift/Sema/ConstraintSystem.h:6249]: Thanks!

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please build toolchain macOS platform

**Comment by xedin** [comment]
@swift-ci please build toolchain macOS platform

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test macOS platform

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please test

**Pull request closed without merging**